bonfire/namespaces.py

```diff
@@ -353,8 +356,11 @@ def describe_namespace(project_name: str):
         raise FatalError(f"namespace '{project_name}' was not reserved with namespace operator")
 
     routes = get_json("route", namespace=project_name)
-    host = get_fe_hostname(routes)
+    if not routes.get('items'):
+        raise FatalError(f"namespace '{project_name}' does not have Routes (is is being"
```

Comment:
I don't think this is the best way to see if a namespace is in the process of getting cleaned up. You could possibly deploy templates that do not result in any Routes being created. Also, Routes are an OpenShift-specific thing and bonfire is technically supposed to work for deployments to both minikube or OpenShift -- on minikube there will be no Routes :)

A more reliable way might be if you see that the namespace is in 'Terminating' state when you look at its kubernetes object 'status' or something like that.

Is this kind of race condition a problem you run into frequently? If so maybe we can address this in a follow-up PR.

Reply:
This fixed a `KeyError` from `get_fe_hostname`. I can keep this out of this PR.

The reproducer is:
# bonfire namespace reserve
# bonfire namespace release
# bonfire namespace describe

Reply:
I've took it out of this PR. We can submit that separately to not block this one.

Reply:
Opened https://github.com/RedHatInsights/bonfire/issues/281 to track